text/0052-RDF-for-public-name-resolution/0052-RDF-for-public-name-resolution.md

```diff
@@ -64,47 +67,117 @@ If so, it is resolved via XOR-URL.
 - If another data type is found, say a `FilesContainer` or `ImmutableData`, that data is retrieved (see point 3 below).
 
 
-#### 2. PublicNameSystem.
+#### 2. NRS-URL.
 
 Failing to be detected as a XOR-URL, we then parse the url and use the Public Name System to resolve for data.
 
-Here the URL terminology for `host` is equivalent to a SAFE `Public Name`. What are known as `subdomains` on the clearnet are referred to as `SubNames`.
+Here the DNS terminology for `domain` is equivalent to a SAFE `Public Name`. What is known as `top level domain` in DNS is referred to as `Top Name`. What are known as `sub domains` in DNS are referred to as `Sub Names`.
 
-`safe://<subName>.<publicName>?v=<version>`
+```
+safe://<subName>.<topName>?v=<version>
+       |-----------------|
+            Public Name
+```			
 
-- GET the AppendOnlyData for a given `Public Name`, at the specified `version`.
+```
+Example:
+  Public Name -->  a.myname
+  Top Name    -->  myname
+  Sub Names   -->  a
+```
+
+```
+Example without subnames:
+  Public Name -->  myname
+  Top Name    -->  myname
+  Sub Names   -->  <empty>
+```
+
+- GET the AppendOnlyData for a given `topName`, at the specified `version`.
 - Parse the retrieved `Resolvable Map`.
-- Resolve the `Sub Name` graph from this `Resolvable Map`.
+- Resolve the `Sub Name` graph / SAFE-URL from this `Resolvable Map`.
 
 Unavailability of any data being dereferenced will throw an error.
 
 ##### 2.1 No `SubName` aka Default Services.
 
-- GET the AppendOnlyData for a given `Public Name`.
+- GET the AppendOnlyData for a given `Top Name`.
 - Parse the retrieved `Resolvable Map`.
-- Resolve the `default` graph / XOR-URL if available.
+- Resolve the `default` graph / SAFE-URL if available.
 
 Unavailability of any data being dereferenced will throw an error.
 
 
 ##### 2.2 Many `SubName`s
 
-`safe://<subName>.<subName>.<subName>.<subName>.<publicName>`
+```
+safe://<subName>.<subName>.<subName>.<subName>.<topName>
+       |-----------------------------------------------|
+		                  Public Name
+```
+
+```
+Example:
+  Public Name -->  a.b.c.d.myname
+  Top Name    -->  myname
+  Sub Names   -->  a.b.c.d
+```
 
 - As above, resolving each additional substring, up to a defined maximum of redirects (implemented in the resolver.)
 	- Safe Browser will implement redirect limit of 10 redirects per url resolution. Any more than this would throw an error.
 
 Unavailability of any data being dereferenced will throw an error.
 
+#### 3. Path resolution
+
+When resolving NRS-URL `link` to SAFE-URL `target`, the path of
+`link` is appended to the path of `target`.
+
+Example: An NRS-URL has been created with `Top Name` test:
+
+```
+safe://test  --> safe://<xor>/testdata
```

Comment:
It may well confuse the point, but a `v` would be needed here. Perhaps if we don't want to include it here, maybe just an asterisk to note it? (As re-reading we're not _that_ explicit about that requirement in this RFC).

Reply:
good point.  I will add it.

Reply:
Ok, I renamed the section to Path and Version resolution and gave rules/examples for both.  Please double-check it is correct/logical.